### PR TITLE
Address-Book

### PR DESCRIPTION
**Pull request by fileninja**
implemented an Address-Book with options to add, delete and modify contacts. to open it, you can click the book symbol in the top menu bar. I've also added an Autocompletion example in the sendArk Function. Just type an username of your added contacts and the text field will fill in the deposited address.

### CODE REVIEW

**Comment by fileninja** [comment]
#73 is done with that.

**Comment by zillionn** [comment]
Wow, great job!

**Comment by fix** [comment]
agree great one!

File: client/app/src/Addressbook/AddressbookController.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    .controller('AddressbookController',
+    ['$scope', '$mdDialog', "$mdToast", "storageService", "gettextCatalog", AddressbookController]);
+
+  function AddressbookController($scope, $mdDialog, $mdToast, storageService, gettextCatalog) {
+
+    var self = this;
+    var contacts;
+
+    self.getContacts = function () {
+      self.contacts = storageService.getGlobal("contacts");
```
Comment (fix) [inline client/app/src/Addressbook/AddressbookController.js]: should not be stored at global level. Use get/set instead

File: client/app/src/accounts/AccountController.js [hunk 2]
```diff
@@ -1000,8 +1000,9 @@
 
       function querySearch(text){
         text=text.toLowerCase();
-        var filter=self.accounts.filter(function(account){
-          return (account.address.toLowerCase().indexOf(text)>-1) || (account.username && (account.username.toLowerCase().indexOf(text)>-1));
+        var contacts = storageService.getGlobal("contacts");
```
Comment (fix) [inline client/app/src/accounts/AccountController.js]: use storageService.get("contacts")

File: client/app/src/Addressbook/AddressbookController.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      if (self.contacts == null || self.contacts == undefined) self.contacts = [];
+    }
+
+    self.getContacts();
+
+    self.save = function () {
+      storageService.setGlobal("contacts", self.contacts);
```
Comment (fix) [inline client/app/src/Addressbook/AddressbookController.js]: use storageService.set(...)

**Comment by fileninja** [comment]
Ah I See.. I will rework it

**Comment by fileninja** [comment]
removed the global storage, so your hints are resolved now

**Comment by zillionn** [comment]
Couple things:
1. How about to change the NEW button to something like this, so it's more distinguishable:
<md-button class="md-button" ng-click="ab.addAddressbookContact()"><md-icon md-font-library="material-icons">library_add</md-icon> <b translate>ADD ADDRESS</b></md-button>

2. Also you should put the NEW/ADD button first, because when you add more addresses you need to scroll to the bottom to see it.

3. What's the point of "md-prevent-menu-close"? It doesn't close when I click NEW.

**Comment by fileninja** [comment]
all of your Points are legitimate. I have to admit that I was more Focused in the functionality.. I will push a new improved commit.

**Comment by fileninja** [comment]
here it is

File: client/app/src/Addressbook/AddressbookController.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        }
+      }
+      return false;
+    }
+
+    self.showToast = function (message, variable) {
+      $mdToast.show(
+        $mdToast.simple()
+          .textContent(gettextCatalog.getString(message) + variable)
+          .hideDelay(5000)
```
Comment (zillionn) [inline client/app/src/Addressbook/AddressbookController.js]: You could add ".theme('error')" if error toast so it will be red.

**Comment by fileninja** [comment]
good point. added the theme and furthermore added an check if the text fields are empty.

**Comment by fix** [comment]
great PR! +5

**Comment by luciorubeens** [comment]
hi @fileninja, i tested your feature today and i liked! i just add some [fixes](https://github.com/ArkEcosystem/ark-desktop/commit/f189f50dc26b968416765b4a6918becef3232188) to better ux.

**Pull request merged**